pkg/model/bootstrapscript.go

```diff
@@ -88,3 +94,67 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 	}
 	return fi.WrapResource(templateResource), nil
 }
+
+func (b *BootstrapScript) createProxyEnv(ps *kops.EgressProxySpec) string {
+	var buffer bytes.Buffer
+
+	if ps != nil && ps.HTTPProxy.Host != "" {
+		var httpProxyUrl string
+
+		// TODO double check that all the code does this
+		// TODO move this into a validate so we can enforce the string syntax
+		if !strings.HasPrefix(httpProxyUrl, "http://") {
+			httpProxyUrl = "http://"
+		}
+
+		if ps.HTTPProxy.Port != 0 {
+			httpProxyUrl += ps.HTTPProxy.Host + ":" + strconv.Itoa(ps.HTTPProxy.Port)
+		} else {
+			// todo should we require port?
+			httpProxyUrl += ps.HTTPProxy.Host
+		}
+
+		// Set base env variables
+		buffer.WriteString("export http_proxy=" + httpProxyUrl + "\n")
+		buffer.WriteString("export https_proxy=${http_proxy}\n")
+		buffer.WriteString("export no_proxy=" + ps.ProxyExcludes + "\n")
+		buffer.WriteString("export NO_PROXY=${no_proxy}\n")
+
+		// TODO move the rest of this configuration work to nodeup
```

Comment:
Yes - I am very wary of doing stuff in the bootstrap script.

I guess the argument here is that we need the proxy to download nodeup.  That makes sense.

We should move the rest of this to nodeup asap though...